src/main/java/org/minions/devfund/angela/battleship/Ship.java

```diff
@@ -0,0 +1,223 @@
+package org.minions.devfund.angela.battleship;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+/**
+ * Manages all Ships.
+ */
+public abstract class Ship {
+
+    private int bowRow;
+    private int bowColumn;
+    private int length;
+    private boolean horizontal;
+    private boolean[] hit;
+
+    /**
+     * Obtains the ship type.
+     *
+     * @return String with the ship type.
+     */
+    abstract String getShipType();
+
+    /**
+     * Verifies if it is possible to place a ship in a position.
+     *
+     * @param row        the row where to place the ship.
+     * @param column     the column where to place the ship.
+     * @param horizontal the orientation to position the ship.
+     * @param ocean      {@link Ocean}.
+     * @return true if is possible to place a ship in the position.
+     */
+    boolean okToPlaceShipAt(final int row, final int column, final boolean horizontal, final Ocean ocean) {
+        try {
+            if (horizontal) {
+                for (int i = row; i < row + length; i++) {
+                    if (ocean.isOccupied(i, column)) {
+                        return false;
+                    }
+                }
+
+            } else {
+                for (int j = column; j < column + length; j++) {
+                    if (ocean.isOccupied(row, j)) {
+                        return false;
+                    }
+                }
+            }
+
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Is really needed this catch Exception ?
or Could you handle the exception with a conditional ?